.github/workflows/build-linux-artifacts.yaml

```diff
@@ -0,0 +1,252 @@
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+
+    - name: apt update
+      run: sudo apt update
+
+    - name: Install required packages
+      run: sudo apt install -y debootstrap
+
+    - name: Download test modules
+      uses: actions/download-artifact@v3
+      with:
+        name: test-modules
+        path: build_tests
+
+    - name: Set up QEMU
+      run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+
+    - name: Build AtomVM using docker
+      timeout-minutes: 15
+      run: |
+        docker run --platform linux/${{ matrix.platform }} --rm -v $PWD:/atomvm -w /atomvm \
+        -e CFLAGS="${{ matrix.cflags }}" -e CXXFLAGS="${{ matrix.cflags }}" \
+        ${{ matrix.docker_image }}:${{ matrix.tag }} /bin/bash -c '
+        ([ -n "${{ matrix.sources }}" ] && echo "${{ matrix.sources }}" > /etc/apt/sources.list || true) &&
+        cat /etc/apt/sources.list &&
+        if test -n "${{ matrix.install_deps }}"; then
+            echo
+            ${{ matrix.install_deps }}
+        else
+            apt update &&
+            apt install -y file gcc g++ binutils cmake make doxygen gperf zlib1g-dev libssl-dev tzdata
+        fi &&
+        file /bin/bash &&
+        uname -a &&
+        cc --version &&
+        ld --version &&
+        ldd --version &&
+        echo $CFLAGS &&
+        echo $CXXFLAGS &&
+        cmake --version &&
+        mkdir -p build &&
+        cd build &&
+        cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo ${{ matrix.cmake_opts }} &&
+        cp ../build_tests/tests/erlang_tests/*.beam tests/erlang_tests/ &&
+        cp ../build_tests/tests/erlang_tests/code_load/*.{avm,beam,hrl} tests/erlang_tests/code_load/ &&
+        mkdir -p tests/erlang_tests/code_load/beams/ &&
+        cp ../build_tests/tests/erlang_tests/code_load/beams/*.beam tests/erlang_tests/code_load/beams/ &&
+        cp ../build_tests/tests/libs/estdlib/*.avm tests/libs/estdlib/  &&
+        cp ../build_tests/tests/libs/eavmlib/*.avm tests/libs/eavmlib/ &&
+        cp ../build_tests/tests/libs/alisp/*.avm tests/libs/alisp/ &&
+        VERBOSE=1 make AtomVM &&
+        make test-erlang &&
+        make test-enif &&
+        make test-mailbox &&
+        make test-structs &&
+        file ./tests/test-erlang &&
+        ./tests/test-erlang -s prime_smp &&
+        file ./tests/test-enif &&
+        ./tests/test-enif &&
+        file ./tests/test-mailbox &&
+        ./tests/test-mailbox &&
+        file ./tests/test-structs &&
+        ./tests/test-structs &&
+        file ./src/AtomVM &&
+        ./src/AtomVM tests/libs/estdlib/test_estdlib.avm &&
+        ./src/AtomVM tests/libs/eavmlib/test_eavmlib.avm &&
+        ./src/AtomVM tests/libs/alisp/test_alisp.avm &&
+        cp ./src/AtomVM ./AtomVM-${{ matrix.build_name }}-${{ github.ref_name }} &&
+        sha256sum ./src/AtomVM > ./AtomVM-${{ matrix.build_name }}-${{ github.ref_name }}.sha256
+        '
+
```

Comment:
Should we also include `PackBEAM`?  It is not as full-featured as the `rebar3` and `mix` plugins, so I am okay with just treating it as an internal tool, for now, and not releasing it in the wild.  It might be too hard to support.

Reply:
Executive decision: no.